SlackAPI/SlackClient.cs

```diff
@@ -52,22 +37,24 @@ public class SlackClient
         public Dictionary<string, DirectMessageConversation> DirectMessageLookup;
         public Dictionary<string, Conversation> ConversationLookup;
 
-        //public event Action<ReceivingMessage> OnUserTyping;
-        //public event Action<ReceivingMessage> OnMessageReceived;
-        //public event Action<ReceivingMessage> OnPresenceChanged;
-        //public event Action<ReceivingMessage> OnHello;
-
         public SlackClient(string token)
         {
             APIToken = token;
         }
 
+        public SlackClient(string token, IWebProxy proxySettings)
+            : base(proxySettings)
+        {
+            APIToken = token;
+        }
+
 		public virtual void Connect(Action<LoginResponse> onConnected = null, Action onSocketConnected = null)
         {
             EmitLogin((loginDetails) =>
             {
-				if(loginDetails.ok)
-					Connected(loginDetails);
+                loginDetails.AssertOk();
```

Comment:
I added this AssertOK to raise an explicit exception in the Connect method if the token is invalid or the proxy settings are incorrect.
Without this AssertOk call, we have a NullReferenceException later because loginDetails is null and the reason of the NRE is less obvious